cpp/include/cuspatial/experimental/point_linestring_distance.cuh

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cuspatial {
+
+/**
+ * @ingroup distance
+ * @copybrief cuspatial::pairwise_point_linestring_distance
+ *
+ * The number of distances computed is `std::distance(points_first, points_last)`.
+ *
+ * @tparam Cart2dItA iterator type for point array of the point element of each pair. Must meet
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam Cart2dItB iterator type for point array of the linestring element of each pair. Must meet
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OffsetIterator iterator type for offset array. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OutputIt iterator type for output array. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ *
+ * @param points_first beginning of the range of points making up the first element of each
+ * pair
+ * @param points_last end of the range of the points making up the first element of each pair
+ * @param linestring_offsets_first beginning of the range of the offsets to the linestring element
+ * of each pair
+ * @param linestring_points_first beginning of the range of points of the linestring element of each
+ * pair
+ * @param linestring_points_last end of the range of points of the linestring element of each pair
+ * @param distances_first beginning the output range of distances
+ * @param stream The CUDA stream to use for device memory operations and kernel launches.
```

Comment:
I think these two docs should distinct more from each other because they serve different purposes. The one in detail name space is documentation on an internal kernel and this one is for users. IMO the doc on the kernel is way too detailed than it needs to be and there contains too much duplication like you mentioned above. I would suggest removing the parameter description part since they are similar to the API description, and only leave essential information describing what the kernel does.

Reply:
You can add docs (e.g. additional @param) after @copydoc, but you can't remove.

Reply:
I should point out the previous documentation is internal docs. It will not show up in the doxygen website. In that case `@copydoc` will have no effect but could confuse developers because redirects them to the function where the doc is copied from, which it may not be evident.

Reply:
> but you can't remove.
Do we want different doc contents for the internal function v.s. the user API? IMO the internal function doc should describe how the function works to help developer understand the logics, and user API docs should focus on usage, parameter constraints, exceptions etc..

Reply:
Personally, having started this conversation, I prefer a single doc. Everything being developed in C++ RAPIDS, `detail`, `experimental`, and public namespaces, is targeted at power users aka advanced developers. I don't see much of a distinction between developer and user. Plus maintenance is much more complicated to maintain parallel docsets that are almost the same but differ in obscure ways. I'm not dogmatic about this though, if you think more refined documentation is appropriate I won't stop you.

Reply:
In fact, since the internal function will not be rendered on doxygen website, `@copydoc` will essentailly have no effect. Eventually the function's docstring goes with the function it decorates. While reducing duplication is a good thing, keeping track of remote referencing could be even a harder task to maintain without the help of doxygen.